src/App.tsx

```diff
@@ -1,26 +1,215 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState, useRef } from 'react';
+import * as postService from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoList } from './components/TodoList';
+import TodoFooter from './components/TodoFooter';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoFilter } from './enums/TodoFilter';
 
-const USER_ID = 0;
+type Filter = 'all' | 'active' | 'completed';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  // eslint-disable-next-line max-len
+  const [loadingTodos, setLoadingTodos] = useState<{ [key: number]: boolean }>(
+    {},
+  );
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isActive] = useState<number>();
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+  const [filter, setFilter] = useState<Filter>(TodoFilter.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const isAnyLoading = Object.values(loadingTodos).some(isLoading => isLoading);
+
+  // #region inputFocus
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const inputfocus = () => {
+    const timer = setTimeout(() => {
+      if (inputRef.current) {
+        inputRef.current.focus();
+      }
+    }, 0);
+
+    return () => clearTimeout(timer);
+  };
+
+  useEffect(() => {
+    inputfocus();
+  }, []);
+  //#endregion
+
+  // #region loadTodos
+  const loadTodos = async () => {
+    setErrorMessage('');
+
+    postService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'));
+  };
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+  //#endregion
+
+  // #region filter
+  const filteredTodos = todos.filter(todo => {
+    if (filter === TodoFilter.ACTIVE) {
+      return !todo.completed;
+    }
+
+    if (filter === TodoFilter.COMPLETED) {
+      return todo.completed;
+    }
+
+    return true;
+  });
+
+  const handleFilterChange = (newFilter: Filter) => {
+    setFilter(newFilter);
+  };
+  //#endregion
+
+  // #region add, delete
+  function addPost(title: string) {
```

Comment:
It is important to follow the same way of creating functions everywhere
```suggestion
  const addPost = (title: string) => {
```